ocaml/lambda/lambda.mli

```diff
@@ -222,34 +222,53 @@ type primitive =
-  | Pstring_load_32 of bool * alloc_mode
-  | Pstring_load_f32 of bool * alloc_mode
-  | Pstring_load_64 of bool * alloc_mode
-  | Pstring_load_128 of { unsafe : bool; mode: alloc_mode }
-  | Pbytes_load_16 of bool
-  | Pbytes_load_32 of bool * alloc_mode
-  | Pbytes_load_f32 of bool * alloc_mode
-  | Pbytes_load_64 of bool * alloc_mode
-  | Pbytes_load_128 of { unsafe : bool; mode: alloc_mode }
-  | Pbytes_set_16 of bool
-  | Pbytes_set_32 of bool
-  | Pbytes_set_f32 of bool
-  | Pbytes_set_64 of bool
-  | Pbytes_set_128 of { unsafe : bool }
+  | Pstring_load_16 of { unsafe : bool; index_kind : array_index_kind }
+  | Pstring_load_32 of { unsafe : bool; index_kind : array_index_kind;
+      mode : alloc_mode; boxed : bool }
+  | Pstring_load_f32 of { unsafe : bool; index_kind : array_index_kind;
+      mode : alloc_mode; boxed : bool }
+  | Pstring_load_64 of { unsafe : bool; index_kind : array_index_kind;
+      mode : alloc_mode; boxed : bool }
+  | Pstring_load_128 of { unsafe : bool; index_kind : array_index_kind;
+      mode: alloc_mode }
+  | Pbytes_load_16 of { unsafe : bool; index_kind : array_index_kind }
+  | Pbytes_load_32 of { unsafe : bool; index_kind : array_index_kind;
+      mode : alloc_mode; boxed : bool }
+  | Pbytes_load_f32 of { unsafe : bool; index_kind : array_index_kind;
+      mode : alloc_mode; boxed : bool }
+  | Pbytes_load_64 of { unsafe : bool; index_kind : array_index_kind;
+      mode : alloc_mode; boxed : bool }
+  | Pbytes_load_128 of { unsafe : bool; index_kind : array_index_kind;
+      mode: alloc_mode }
+  | Pbytes_set_16 of { unsafe : bool; index_kind : array_index_kind }
+  | Pbytes_set_32 of { unsafe : bool; index_kind : array_index_kind;
+      boxed : bool }
+  | Pbytes_set_f32 of { unsafe : bool; index_kind : array_index_kind;
+      boxed : bool }
+  | Pbytes_set_64 of { unsafe : bool; index_kind : array_index_kind;
+      boxed : bool }
+  | Pbytes_set_128 of { unsafe : bool; index_kind : array_index_kind;
+      boxed : bool }
   (* load/set 16,32,64,128 bits from a
      (char, int8_unsigned_elt, c_layout) Bigarray.Array1.t : (unsafe) *)
-  | Pbigstring_load_16 of { unsafe : bool }
-  | Pbigstring_load_32 of { unsafe : bool; mode: alloc_mode; boxed : bool }
-  | Pbigstring_load_f32 of { unsafe : bool; mode: alloc_mode; boxed : bool }
-  | Pbigstring_load_64 of { unsafe : bool; mode: alloc_mode; boxed : bool }
-  | Pbigstring_load_128 of { aligned : bool; unsafe : bool; mode: alloc_mode;
+  | Pbigstring_load_16 of { unsafe : bool; index_kind : array_index_kind }
+  | Pbigstring_load_32 of { unsafe : bool; index_kind : array_index_kind; mode :
+      alloc_mode; boxed : bool }
+  | Pbigstring_load_f32 of { unsafe : bool; index_kind : array_index_kind;
+      mode : alloc_mode; boxed : bool }
+  | Pbigstring_load_64 of { unsafe : bool; index_kind : array_index_kind; mode :
+      alloc_mode; boxed : bool }
+  | Pbigstring_load_128 of { aligned : bool; unsafe : bool; index_kind :
```

Comment:
Some of this formatting seems weird.  Maybe take this type declaration and paste it into an autoformatted file, format it, then paste just this section back?

Reply:
`ocamlformat` puts each record field onto its own line. I can adopt this format if we prefer, but it doesn't seem to match the rest of the file. So for now I've accepted `ocamlformat`'s output, but combined record fields onto one line when possible (without splitting on `:`).